src/runtime.c

```diff
@@ -577,6 +577,24 @@ int main(int argc, char *argv[]) {
 #endif
     }
 
+    // calculate full path of AppImage
+    char appimage_fullpath[PATH_MAX];
+
+    if(getenv("TARGET_APPIMAGE") == NULL) {
```

Comment:
This complexity should be moved into a function.

I don't quite see why `TARGET_APPIMAGE` must be evaluated more than once. Calling `readlink` or `realpath` on an absolute path is completely harmless, and cheap enough to consider this optimization unnecessary. I'd just always resolve.

Reply:
I kept the complexity here because, as I explained earlier, `readlink` and `realpath` can give different values in the case of nested symlinks, and I didn't think that this PR was the right place to make that change.

After a closer look though, it looks like I could just get rid of `appimage_path` and just use `appimage_full_path` everywhere, since the only time they would be different with this PR is when `$TARGET_APPIMAGE` is set.  It looks like reading the AppImage always uses `appimage_path` (which is _unresolved_ with `$TARGET_APPIMAGE`), so changing that to be the resolved path should still be okay.  I still want to keep using `realpath` in this code path though, since users may be using portable home/config directories with appimaged and I don't want to change how those AppImage paths are resolved.